### PR TITLE
Protecting string converters

### LINKED ISSUE #20: Catch exceptions in std::stod
In utils.cpp, we should handle invalid arguments (for steps and unitcell parsers) using a try{}catch(invalid_argument){}.
For now, in case of wrong argument (eg --cell=sdt:yr:4) the program gives: "Error stod"

### PR DESCRIPTION
**Pull request by lscalfi**
Creating `string2double`, `string2long` and `string2ll` functions for `stod`, `stol` and `stoll` catching `std::invalid_argument` exceptions.

Fix #20

### CODE REVIEW

**Comment by Luthaf** [comment]
I agree with the `string2long` function, but the `string2ll` is only used in the `steps_range` parsing. I don't think we need another public function for that, we should just wrap the `steps_range::parse` in a `try {} catch {}` block.

File: src/utils.cpp [hunk 1]
```diff
@@ -120,7 +122,12 @@ steps_range steps_range::parse(const std::string& string) {
 
         if (splitted.size() == 3) {
             if (splitted[2] != "") {
-                auto stride = std::string2ll(splitted[2]);
+                long long stride = -1;
+                try{
```
Comment (Luthaf) [inline src/utils.cpp]: A small bit of formatting: `try {` (with a space). 

FYI, you can wrap multiple statements in a try/catch block:

```cpp
try {
    var = foo();
    do_stuff(var, baz(45));
} catch (...) {
     // Get here if any operation throws
}
```

**Comment by Luthaf** [comment]
Thanks! Now anther step in the git madness, could you squash your commits? See https://robots.thoughtbot.com/git-interactive-rebase-squash-amend-rewriting-history for more information about squashing.

**Comment by lscalfi** [comment]
I think I did it!

**Comment by Luthaf** [comment]
Merged in https://github.com/chemfiles/cfiles/commit/12edfd061517f5d4231df6484cf06b0fffabb28f

(I tried to fix conflicts by hand, but it looks like it confused github)

**Pull request closed without merging**